core/src/main/java/feast/core/service/ProjectService.java

```diff
@@ -16,35 +16,58 @@
  */
 package feast.core.service;
 
+import feast.core.auth.authorization.AuthorizationProvider;
+import feast.core.auth.authorization.AuthorizationResult;
+import feast.core.config.FeastProperties;
+import feast.core.config.FeastProperties.SecurityProperties;
 import feast.core.dao.ProjectRepository;
 import feast.core.model.Project;
 import java.util.List;
 import java.util.Optional;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.ObjectProvider;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
 import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Transactional;
 
 @Slf4j
 @Service
-public class AccessManagementService {
+public class ProjectService {
+
+  private SecurityProperties securityProperties;
+  private AuthorizationProvider authorizationProvider;
   private ProjectRepository projectRepository;
 
+  public ProjectService(
+      FeastProperties feastProperties,
+      ProjectRepository projectRepository,
+      AuthorizationProvider authorizationProvider) {
+    this.projectRepository = projectRepository;
+    this.authorizationProvider = authorizationProvider;
+    this.securityProperties = feastProperties.getSecurity();
+  }
+
   @Autowired
-  public AccessManagementService(ProjectRepository projectRepository) {
+  public ProjectService(
+      FeastProperties feastProperties,
+      ProjectRepository projectRepository,
+      ObjectProvider<AuthorizationProvider> authorizationProvider) {
     this.projectRepository = projectRepository;
     // create default project if it does not yet exist.
     if (!projectRepository.existsById(Project.DEFAULT_NAME)) {
       this.createProject(Project.DEFAULT_NAME);
     }
+    this.authorizationProvider = authorizationProvider.getIfUnique();
+    this.securityProperties = feastProperties.getSecurity();
   }
 
   /**
    * Creates a project
    *
    * @param name Name of project to be created
    */
-  @Transactional
```

Comment:
Why is `@Transactional` removed here? Calling a `projectRepository` writes to the database which want to be atomic.

Reply:
@woop again, If it's not your change should be reverted. Let me know.

Reply:
This could have been a change from my side. Let's revert it.